### PR TITLE
Add netflow to fips mode + update logs_dd_url instead of dd_url for ndm products

### PR DESCRIPTION
**Pull request by Kaderinho**
### What does this PR do?

Add support for NDM SNMP Traps and Netflow in FIPS mode

### Additional Notes

Using port `15` because `14` is used by compliance https://github.com/DataDog/datadog-agent/pull/20230/files

### Describe how to test/QA your changes

Set `fips.enabled: true` + set `network_devices.metadata.dd_url`, `network_devices.snmp_traps.forwarder.dd_url` and `network_devices.netflow.forwarder.dd_url` to a dummy value in `datadog.yaml` and run the Agent. 

Run `datadog-agent config` and verify that `network_devices.metadata.logs_dd_url`, `network_devices.snmp_traps.forwarder.logs_dd_url` and `network_devices.netflow.forwarder.logs_dd_url` are overwritten (to `localhost:9811`, `localhost:9812` and `localhost:9818` resp)

Then run `datadog-agent diagnose --include  connectivity-datadog-event-platform -v` and verify that NDM related diagnose have these in their output (dummy values of *.dd_url should not appear)

- `Connectivity to https://localhost:9811/api/v2/ndm`
- `Connectivity to https://localhost:9812/api/v2/ndmtraps`
- `Connectivity to https://localhost:9818/api/v2/ndmflow`

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: releasenotes/notes/add-ndm-to-fips-mode-f0b6e379369c3f48.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Each section from every release note are combined when the
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+features:
```
Comment (estherk15) [inline releasenotes/notes/add-ndm-to-fips-mode-f0b6e379369c3f48.yaml:9]: Doublechecking if the formatting is the same between   `  - |` and `-`
I saw other PRs formatted like the following:
```
features:
  - |
    Add FIPS mode support for Network Device Monitoring products
```

**Review by estherk15 — APPROVED** [review]

File: releasenotes/notes/add-ndm-to-fips-mode-f0b6e379369c3f48.yaml (hunk 1 above)
Comment (Kaderinho) [reply releasenotes/notes/add-ndm-to-fips-mode-f0b6e379369c3f48.yaml:9]: If I'm not mistaken,`|` is used to write multi lines strings in YAML so I believe both are working correctly in this specific case

**Review by jmw51798 — APPROVED** [review]

File: pkg/config/config.go [hunk 2]
```diff
@@ -1654,6 +1655,7 @@ func setupFipsEndpoints(config Config) error {
 		appsecEvents               = 11
 		orchestratorExplorer       = 12
 		runtimeSecurity            = 13
+		networkDevicesNetflow      = 15
```
Comment (sgnn7) [inline pkg/config/config.go:1658]: not a blocker and just out of curiosity: why the skip of 14 (could we add a comment about it)?

**Review by sgnn7 — APPROVED** [review]

File: pkg/config/config.go (hunk 2 above)
Comment (Kaderinho) [reply pkg/config/config.go:1658]: Because 14 is used in this PR https://github.com/DataDog/datadog-agent/pull/20230/files which will be merged soon 🤞

**Review by NouhaManai96 — APPROVED** [review]

**Review by NouhaManai96 — APPROVED** [review]

**Review by jmw51798 — APPROVED** [review]

**Pull request merged**